xonsh/built_ins.py

```diff
@@ -436,9 +441,15 @@ def run_subproc(cmds, captured=True):
             stdin = PIPE
         else:
             stdin = prev_proc.stdout
+        def subproc_pre():
```

Comment:
This doesn't seem like it needs to be defined inside of `run_subproc()` though I could be missing something subtle :)